octoprint_enclosure/__init__.py

```diff
@@ -1346,6 +1403,55 @@ def read_max31855_temp(self, address):
             self.log_error(ex)
             return 0
 
+
+
+    def handle_emc_linked_temperature(self):
+        try:
+            for pwm_output in list(filter(lambda item: item['output_type'] == 'emc',
+                                          self.rpi_outputs)):
+                if self._printer.is_printing():
+                    index_id = self.to_int(pwm_output['index_id'])
+                    linked_id = self.to_int(pwm_output['linked_temp_sensor'])
+                    linked_data = self.get_linked_temp_sensor_data(linked_id)
+                    current_temp = self.to_float(linked_data['temperature'])
+
+                    duty_a = self.to_float(pwm_output['duty_a'])
+                    duty_b = self.to_float(pwm_output['duty_b'])
+                    temp_a = self.to_float(pwm_output['temperature_a'])
+                    temp_b = self.to_float(pwm_output['temperature_b'])
+
+                    try:
+                        calculated_duty = ((current_temp - temp_a) * (duty_b - duty_a) / (temp_b - temp_a)) + duty_a
+
+                        if current_temp < temp_a:
+                            calculated_duty = 0
+                    except:
+                        calculated_duty = 0
+
+                    self._logger.debug("Calculated duty for EMC %s is %s", index_id, calculated_duty)
+                elif self.print_complete:
+                    calculated_duty = self.to_int(pwm_output['default_duty_cycle'])
+                else:
+                    calculated_duty = self.to_int(pwm_output['default_duty_cycle'])
+            script = os.path.dirname(os.path.realpath(__file__)) + "/SETEMC2101.py"
+            cmd = [sys.executable, script, str(int(calculated_duty))]

```

Comment:
`calculated_duty` remains undefined if you have not added an emc output, because the `for` loop does not perform any iterations.
This causes the following crash:
```
octoprint.plugins.enclosure - WARNING - An exception of type UnboundLocalError occurred on log_error. Arguments:
("local variable 'calculated_duty' referenced before assignment",)
Traceback (most recent call last):
  File "/home/pi/oprint/lib/python3.9/site-packages/octoprint_enclosure/__init__.py", line 1461, in handle_emc_linked_temperature
    cmd = [sys.executable, script, str(int(calculated_duty))]
UnboundLocalError: local variable 'calculated_duty' referenced before assignment
```